### PR TITLE
create a new cache status RWW_HIT

### PR DESCRIPTION
**Pull request by vmamidi**
This PR creates a cache status when the requested object is served to the client while cache is being updated by another request. 
cache status( log field chm) is RWW_HIT
log code is(log field crc) TCP_CF_HIT (https://wiki.squid-cache.org/SquidFaq/SquidLogs)
stat is proxy.process.http.cache_hit_rww

### CODE REVIEW

File: proxy/hdrs/HTTP.h [hunk 1]
```diff
@@ -159,7 +159,8 @@ enum SquidLogCode {
   SQUID_LOG_ERR_FUTURE_1              = 'I',
   SQUID_LOG_ERR_CLIENT_READ_ERROR     = 'J', // Client side abort logging
   SQUID_LOG_ERR_LOOP_DETECTED         = 'K', // Loop or cycle detected, request came back to this server
-  SQUID_LOG_ERR_UNKNOWN               = 'Z'
+  SQUID_LOG_ERR_UNKNOWN               = 'Z',
+  SQUID_LOG_TCP_CF_HIT                = 'F' // Collapsed forwarding HIT also known as Read while write hit
```
Comment (masaori335) [inline proxy/hdrs/HTTP.h:163]: [nitpicking] why only this is named `CF_HIT`? For consistency with something?

Comment (vmamidi) [reply proxy/hdrs/HTTP.h:163]: Yes, it is named according to squid format. 
https://wiki.squid-cache.org/SquidFaq/SquidLogs

**Comment by masaori335** [comment]
[approve ci]

**Comment by vmamidi** [comment]
[approve ci]

**Comment by bneradt** [comment]
[approve ci]

File: proxy/hdrs/HTTP.h (hunk 1 above)
Comment (mlibbey) [reply proxy/hdrs/HTTP.h:163]: CF --> "Collapsed Forwarding"

Comment (randall) [reply proxy/hdrs/HTTP.h:163]: Looks like this list is sorted alphabetically by value , the new option should be moved up

Comment (vmamidi) [reply proxy/hdrs/HTTP.h:163]: @randall yeah, i will update that.

**Comment by vmamidi** [comment]
[approve ci]

**Comment by bneradt** [comment]
[approve ci]

**Comment by bneradt** [comment]
[approve ci]

**Comment by bneradt** [comment]
[approve ci]

**Comment by zwoop** [comment]
[approve ci].

**Comment by zwoop** [comment]
[approve ci].

**Comment by ezelkow1** [comment]
[approve ci]

**Comment by ezelkow1** [comment]
[approve ci]

**Comment by mlibbey** [comment]
@djcarlin think you were asking for this a few summits ago :)

File: proxy/logging/Log.cc [hunk 2]
```diff
@@ -613,7 +613,7 @@ Log::init_fields()
 
   Ptr<LogFieldAliasTable> cache_code_map = make_ptr(new LogFieldAliasTable);
   cache_code_map->init(
-    52, SQUID_LOG_EMPTY, "UNDEFINED", SQUID_LOG_TCP_HIT, "TCP_HIT", SQUID_LOG_TCP_DISK_HIT, "TCP_DISK_HIT", SQUID_LOG_TCP_MEM_HIT,
+    53, SQUID_LOG_EMPTY, "UNDEFINED", SQUID_LOG_TCP_HIT, "TCP_HIT", SQUID_LOG_TCP_DISK_HIT, "TCP_DISK_HIT", SQUID_LOG_TCP_MEM_HIT,
```
Comment (shinrich) [inline proxy/logging/Log.cc:616]: Should we have a constant for the 53?

**Review by SolidWallOfCode — APPROVED** [review]

**Pull request merged**